src/Catalyst.Core.Modules.Ledger/Ledger.cs

```diff
@@ -256,5 +292,117 @@ public void Dispose()
+            }
+
+            public bool IsTracingActions => false;
+            public bool IsTracingOpLevelStorage => false;
+            public bool IsTracingMemory => false;
+            public bool IsTracingInstructions => false;
+            public bool IsTracingCode => false;
+            public bool IsTracingStack => false;
+            public bool IsTracingState => false;
+            public void ReportBalanceChange(Address address, UInt256? before, UInt256? after) { throw new NotImplementedException(); }
+            public void ReportCodeChange(Address address, byte[] before, byte[] after) { throw new NotImplementedException(); }
+            public void ReportNonceChange(Address address, UInt256? before, UInt256? after) { throw new NotImplementedException(); }
+            public void ReportStorageChange(StorageAddress storageAddress, byte[] before, byte[] after) { throw new NotImplementedException(); }
+            public void StartOperation(int depth, long gas, Instruction opcode, int pc) { throw new NotImplementedException(); }
+            public void ReportOperationError(EvmExceptionType error) { throw new NotImplementedException(); }
+            public void ReportOperationRemainingGas(long gas) { throw new NotImplementedException(); }
+            public void SetOperationStack(List<string> stackTrace) { throw new NotImplementedException(); }
+            public void ReportStackPush(Span<byte> stackItem) { throw new NotImplementedException(); }
+            public void SetOperationMemory(List<string> memoryTrace) { throw new NotImplementedException(); }
+            public void SetOperationMemorySize(ulong newSize) { throw new NotImplementedException(); }
+            public void ReportMemoryChange(long offset, Span<byte> data) { throw new NotImplementedException(); }
+            public void ReportStorageChange(Span<byte> key, Span<byte> value) { throw new NotImplementedException(); }
+            public void SetOperationStorage(Address address, UInt256 storageIndex, byte[] newValue, byte[] currentValue) { throw new NotImplementedException(); }
+            public void ReportSelfDestruct(Address address, UInt256 balance, Address refundAddress) { throw new NotImplementedException(); }
+
+            public void ReportAction(long gas,
+                UInt256 value,
+                Address @from,
+                Address to,
+                byte[] input,
+                ExecutionType callType,
+                bool isPrecompileCall = false)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void ReportActionEnd(long gas, byte[] output) { throw new NotImplementedException(); }
+            public void ReportActionError(EvmExceptionType evmExceptionType) { throw new NotImplementedException(); }
+            public void ReportActionEnd(long gas, Address deploymentAddress, byte[] deployedCode) { throw new NotImplementedException(); }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'ReportActionEnd'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)